sensu/plugins/metrics-nova-project-quotas.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+
+import sys
+import time
+import argparse
+import traceback
+from os import environ
+from subprocess import Popen, PIPE
+
+import shade
+import novaclient
+import os_client_config
+from novaclient import client
+from keystoneauth1.session import Session
+
+OK       = 0
+WARNING  = 1
+CRITICAL = 2
+UNKNOWN  = 3
+
+
+def source(script, update=1):
+    pipe = Popen("source %s; env" % script, stdout=PIPE, shell=True)
+    data = pipe.communicate()[0]
+
+    env = dict((line.split("=", 1) for line in data.splitlines()))
+    if update:
+        environ.update(env)
+
+    return env
+
+
+class CloudMetrics(object):
+    def __init__(self):
+        self.cloud = shade.openstack_cloud()
+        self.nova = None
+        self._setup_novaclient()
+
+    def _setup_novaclient(self):
+        config = os_client_config.OpenStackConfig()
+        cloud_config = config.get_one_cloud()
+        session = Session(cloud_config.get_auth())
+        self.nova = client.Client("2.1", session=session)
```

Comment:
with shade, once you have a cloud, you can get access to `cloud.nova_client` to do client things. You shouldn't have to import nova client yourself and set up the client object yourself. Just let shade do it.